src/Elliptic/Systems/Xcts/Events/ObserveAdmIntegrals.cpp

```diff
@@ -141,9 +143,10 @@ void local_adm_integrals(
           definite_integral(contracted_linear_momentum_integrand.get(I) *
                                 get(conformal_area_element),
                             face_mesh);
-      center_of_mass->get(I) += definite_integral(
-          center_of_mass_integrand.get(I) * get(conformal_area_element),
-          face_mesh);
+      center_of_mass->get(I) +=
+          definite_integral(contracted_center_of_mass_integrand.get(I) *
+                                get(conformal_area_element),
```

Comment:
I think for Gauss' law you have to either compute derivatives of the conformal unit normal correctly ($\partial_i \bar{n}^j$), which needs second derivatives of the coordinate maps (seems complicated). Or you compute this integral assuming flat space and $n^i=x^i/r$ (seems easier). To do Gauss' law for this, use $n^i=x^i/r$ when computing the integrands and then integrate both the surface and volume with the Euclidean area/volume element. I think I was wrong suggesting to use the conformal face normal here before.

Reply:
Sounds good. For now, I'm only taking the infinite surface integral, but I'm doing it with an Euclidean area element to indicate what we'll do later.